repl/src/main/scala/com/cloudera/livy/repl/Session.scala

```diff
@@ -151,8 +151,8 @@ class Session(
           statement.compareAndTransit(StatementState.Cancelling, StatementState.Cancelled)
         } else {
           _sc.foreach(_.cancelJobGroup(statementId.toString))
+          Thread.sleep(livyConf.getTimeAsMs(RSCConf.Entry.JOB_CANCEL_TRIGGER_INTERVAL))
```

Comment:
I think in most of the case code will jump into this `if` (statementState is `Cancelling`). Because `cancelJobGroup`  is an async operation, the return of this function doesn't mean job is cancelled, mostly it will still be `Cancelling`. So IMO it is not so necessary to add this `if` to check.